packages/gluestick/src/commands/compareVersions/compareModuleVersions.js

```diff
@@ -0,0 +1,73 @@
+/* @flow */
+import type { Logger } from '../../types.js';
+
+const semver = require('semver');
+const chalk = require('chalk');
+const path = require('path');
+
+const packageName = '/package.json';
+
+/**
+ * Opens the project's package.json file and compares version requirements to
+ * the actual version of each dependency and devDependency. It lists the
+ * node_modules that don't satisfy the requirement or are missing and suggests
+ * running `npm install` to amend the issue.
+ *
+ * Corner cases: there are no dependencies and/or devDependencies, the version
+ * is a path to a file, the input file to the package.json or node_modules dir
+ * is either undefined or incorrect.
+ *
+ * Return - returns array of dependencies that are missing or out of date.
+ */
+const compareModuleVersions = (projectPackage: Object, modulePath: string, logger: Logger) => {
+  const discrepancies = [];
+  let devDepDifference = false;
+  let allDependencies;
+
+  if (!projectPackage.dependencies || !projectPackage.devDependencies) {
+    if (!projectPackage.dependencies && !projectPackage.devDependencies) {
+      return discrepancies;
+    }
+    allDependencies = projectPackage.dependencies || projectPackage.devDependencies;
+  } else {
+    const totalNumDependencies = Object.keys(projectPackage.dependencies).length +
+      Object.keys(projectPackage.devDependencies).length;
+    allDependencies = { ...projectPackage.dependencies, ...projectPackage.devDependencies };
+    if (Object.keys(allDependencies).length < totalNumDependencies) {
+      devDepDifference = true;
+    }
+  }
+
+  Object.keys(allDependencies).forEach(dep => {
+    const tempPath = path.join(modulePath, dep, packageName);
+    let tempPackage;
+    try {
+      tempPackage = require(tempPath);
+    } catch (e) {
+      logger.error('The node_module ', `${chalk.yellow(dep)}`, ' does not exist.');
+      const name = ' '.concat(dep);
+      discrepancies.push(name);
+      return;
+    }
+
+    if (allDependencies[dep] && !semver.satisfies(tempPackage.version, allDependencies[dep])) {
+      const fileVersionMatch = allDependencies[dep].match(/\d+\.\d+\.\d+/);
+      if (tempPackage._from.search(fileVersionMatch) === -1) {
```

Comment:
How risky is it to use `_from`? If one of the underscored variables aren't accessed then there's no way to check those node_modules versions and they'll be skipped.